internal/internal_event_handlers.go

```diff
@@ -1379,3 +1383,114 @@ func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecut
 
 	return nil
 }
+
+func (weh *workflowExecutionEventHandlerImpl) estimateHistorySize(event *m.HistoryEvent) int {
+	sum := historySizeEstimationBuffer
+	switch event.GetEventType() {
```

Comment:
I don't think that's viable because all EventAttribute struct are auto-generated code using idl files. We are however able to add history size field into HistoryEvent struct but that makes no difference and are overly complicate things.

Reply:
oh if they are auto generated then this wouldn't work. in that case try to increase the coverage in your size calculations so we are not reporting a total size that's orders of magnitude smaller than actual.

Reply:
I have also added a buffer to act as a catch all for all those smaller fields.

Reply:
it doesn't look like 400 fixed buffer will cover those uncaptured field sizes when history has thousands of events. can you download a long running workflow history and compare your estimate with actual size?

Reply:
400 is per event based. So I think that should be enough. Also, right now I have metrics for the server side history size as well so we can compare after merging and later determine the best number from there

Reply:
And that is going to be configurable during worker startup

Reply:
good point. 400B buffer per event should be good enough